modules/pricing/state_and_plan_content_from_json.tf

```diff
@@ -6,7 +6,15 @@ locals {
   # State - all resources are linear
   # Plan - combine root_module.resources and root_module.child_modules
   content_resources_state = local.input_type == "state" ? jsonencode(local.content.resources) : jsonencode({})
-  content_resources_plan  = local.input_type == "plan" ? jsonencode(concat(lookup(local.content.planned_values.root_module, "resources", []), flatten(concat([for m in lookup(local.content.planned_values.root_module, "child_modules", {}) : m.resources])))) : jsonencode({})
+
+  # local.content.planned_values includes more changes than what will actually occur.
+  # Instead query 'resource_changes' and filter by the actions that will be taken
+  content_resources_plan = local.input_type == "plan" ? jsonencode(
```

Comment:
From what I see this change ignores the changes under `child_modules` which is a very important part.

Reply:
The resources_changes key should include all changes, I believe.

Reply:
Added an example and test fixture:  The output looks like 

```
Changes to Outputs:
  + pricing_plan_input_resources         = {}
  + pricing_plan_pricing_per_resources   = {
      + aws_ebs_volume.submodule1-ebsvol.module.asubmodule.aws_ebs_volume.submodule1-ebsvol[0] = 0.11
      + aws_instance.root-instance.aws_instance.root-instance[0]                               = 0.0208
      + aws_instance.submodule1-instance.module.asubmodule.aws_instance.submodule1-instance[0] = 0.0208
    }
  + pricing_plan_pricing_product_filters = {}
  + pricing_plan_resource_quantity       = {
      + aws_ebs_volume.submodule1-ebsvol.module.asubmodule.aws_ebs_volume.submodule1-ebsvol[0] = 0.0136986301
      + aws_instance.root-instance.aws_instance.root-instance[0]                               = 1
      + aws_instance.submodule1-instance.module.asubmodule.aws_instance.submodule1-instance[0] = 1
    }
  + pricing_plan_resources               = {}
  + pricing_plan_total_price_per_hour    = 0.0431068493
  + pricing_plan_total_price_per_month   = 31.47

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.
```
Which indicates that the submodule data is in the resource_changes key.  You can take a look at the plan.json file as well to see this in effect.